app/interactors/schedule/schedule_interactor.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Schedule::ScheduleInteractor
+  include Interactor
+  delegate :params,
+           :user,
+           :edition,
+           :issues,
+           to: :context
+
+  def call
+    Edition.transaction do
+      find_and_lock_edition
+      check_for_issues
+      schedule_publishing
+      create_scheduling
+    end
+  end
+
+private
+
+  def find_and_lock_edition
+    context.edition = Edition.lock.find_current(document: params[:document])
+  end
+
+  def check_for_issues
+    unless edition.schedulable?
+      # FIXME: this shouldn't be an exception but we've not worked out the
+      # right response - maybe bad request or a redirect with flash?
+      raise "Can't schedule an edition which isn't schedulable"
+    end
+
+    if params[:review_status].blank?
+      issues = Requirements::CheckerIssues.new([
+        Requirements::Issue.new(:scheduling, :not_selected),
```

Comment:
I wonder if this should have a more descriptive naming like "schedule_review" as scheduling feels quite broad

Reply:
Done